### PR TITLE
Feature/ahti 250

### PR DESCRIPTION
**Pull request by marharyta**
Used clustering approach from here https://www.leighhalliday.com/mapbox-clustering
I added the Description to comments

### CODE REVIEW

**Comment by ahie** [comment]
When you commit the final version, remember to update the test snapshot.

**Comment by marharyta** [comment]
Looks like I figured what was the problem, https://codesandbox.io/s/q-57240777-useref-3t17j

**Comment by marharyta** [comment]
Just realized I need to update a snapshot

**Comment by ahie** [comment]
The diff still contains tons of files from develop. Could you rebase your changes on top of the develop branch?

**Comment by marharyta** [comment]
will do

File: src/common/ui-components/Map/Map.module.scss [hunk 1]
```diff
@@ -13,3 +13,13 @@
     margin: $margin;
   }
 }
+
+.cluster-marker {
+  color: #fff;
+  background: #1978c8;
```
Comment (ahie) [inline src/common/ui-components/Map/Map.module.scss]:
```suggestion
  background: var(--hds-brand-color-coat-of-arms-blue);
```

File: src/common/ui-components/Map/Map.tsx [hunk 2]
```diff
@@ -71,8 +112,39 @@ const Map: React.FC<MapProps> = ({ className, features, onClick }) => {
...
+        // this is temporary, new designs should come soon
+        if (isCluster) {
+          return (
+            <Marker
+              key={`cluster-${cluster.id}`}
+              latitude={latitude}
+              longitude={longitude}
+            >
+              <div
+                className="cluster-marker"
```
[the remark this answers is not in the document]
Comment (marharyta) [reply src/common/ui-components/Map/Map.tsx]: @ahie thanks a lot for pointing it out, missed that one

File: src/common/ui-components/Map/Map.tsx [hunk 3]
```diff
@@ -63,6 +70,40 @@ const Map: React.FC<MapProps> = ({ className, features, onClick }) => {
...
+        .getBounds()
+        .toArray()
+        .flat()
+    : null;
+
+  const { clusters } = useSupercluster({
+    points,
+    bounds,
+    zoom: viewPort.zoom,
+    options: { radius: 75, maxZoom: 20 }
```
[the remark this answers is not in the document]
Comment (marharyta) [reply src/common/ui-components/Map/Map.tsx]: Sounds like a plan

[the remark this answers is not in the document]
Comment (marharyta) [reply src/common/ui-components/Map/Map.tsx]: Should we just leave it for now?

[the remark this answers is not in the document]
Comment (marharyta) [reply src/common/ui-components/Map/Map.tsx]: I am not a huge fan of 'magic numbers' myself, but the clustering behaviour is still a subject to design

[the remark this answers is not in the document]
Comment (marharyta) [reply src/common/ui-components/Map/Map.tsx]: @ahie ok fine

**Comment by marharyta** [comment]
@ahie you are right, it doe fail

**Comment by marharyta** [comment]
I am pretty sure it is due to some code mistake though

**Comment by badrange** [comment]
In kukkuu we had solved a CI build failure by clearing Travis' cache of our project. Pretty wild.

File: src/common/ui-components/Map/Map.tsx [hunk 4]
```diff
@@ -112,47 +132,118 @@ const Map: React.FC<MapProps> = ({
...
+              longitude={longitude}
+            >
+              <div
+                className={styles.clusterMarker}
+                style={{
+                  width: `${10 + (pointCount / points.length) * 20}px`,
+                  height: `${10 + (pointCount / points.length) * 20}px`
+                }}
+              >
+                {pointCount}
```
Comment (ahie) [inline src/common/ui-components/Map/Map.tsx]: It looks like there's something funny going on with the clusters when two clusters are close together? Seems to be caused by numbers with more than one digit getting wrapped.

File: src/common/ui-components/Map/Map.tsx [hunk 5]
```diff
@@ -96,12 +116,12 @@ const Map: React.FC<MapProps> = ({
...
       });
     };
-
+    const feature = features.find(feature => feature.id === pointFeature.id);
     return (
       <Marker
-        key={id}
-        longitude={feature.geometry.coordinates[0]}
-        latitude={feature.geometry.coordinates[1]}
+        key={`id_${Math.random()}`}
```
Comment (ahie) [inline src/common/ui-components/Map/Map.tsx]: > `Math.random`

Not guaranteed to be unique. Why not use the id passed in as an argument?

File: src/common/ui-components/Map/Map.tsx (hunk 4 above)
Comment (marharyta) [reply src/common/ui-components/Map/Map.tsx]: Can you record it in a gif? Cannot reproduce it.

**Pull request merged**